src/AbstractRequest.js

```diff
@@ -43,12 +48,17 @@ export default class AbstractRequest {
 
         // Parse promise if need be
         let promise = fetch(endpoint, requestOptions);
+
         if (requestOptions.method === 'DELETE') {
             promise = promise.then(::this.checkStatus);
         } else {
             promise = promise.then(::this.parseJSON);
```

Comment:
This will not work. The `response` our middleware receives would already be a parsed response, instead of the direct result of `fetch`